From 305bc4bc25962d1a555bc1c35dd27fd6c90f5dd6 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Tue, 8 Jul 2008 17:26:48 +0800
Subject: [PATCH 010/280] ENGR00083200: Add support interleaved Cr/Cb map mode
 for FrameBuffer on MX37

Marley VPU supports two memory map modes for the
chrominance components in the frame buffers. One is the separate Cb/Cr map mode,
another is the interleaved Cb/Cr map mode that has only one base address for
interleaved Cb and Cr samples such as Cb0, Cr0, Cb1, Cr1, Cb2, Cr2,..., CbN, CrN, not
Cb0, Cb1, Cb2, ..., CbN, Cr0, Cr1, Cr2, ..., CrN. CbN and CrN are the last sample of
each chrominance component of the decoded frame.

This patch adds support interleaved CrCb map mode, it should be set same as
CrCb setting in V4L for correct display.

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 vpu/vpu_lib.c | 20 +++++++++-----------
 1 file changed, 9 insertions(+), 11 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index ecde115..43bb396 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -1565,7 +1565,7 @@ RetCode vpu_DecGetInitialInfo(DecHandle handle, DecInitialInfo * info)
 	val = ((pDecInfo->dynamicAllocEnable << 3) & 0x8) |
 		((pDecInfo->filePlayEnable << 2) & 0x4) |
 		((pDecInfo->openParam.reorderEnable << 1) & 0x2);
-	
+
 	if (cpu_is_mx27()) {
 		val |= (pDecInfo->openParam.qpReport & 0x1);
 	} else {
@@ -1574,18 +1574,16 @@ RetCode vpu_DecGetInitialInfo(DecHandle handle, DecInitialInfo * info)
 
 	VpuWriteReg(CMD_DEC_SEQ_OPTION, val);
 
-	if (cpu_is_mxc30031()) {
-		VpuWriteReg(BIT_FRAME_MEM_CTRL,
+	if (pCodecInst->codecMode == AVC_DEC) {
+		VpuWriteReg(CMD_DEC_SEQ_PS_BB_START,
+				pDecInfo->openParam.psSaveBuffer);
+		VpuWriteReg(CMD_DEC_SEQ_PS_BB_SIZE,
+			(pDecInfo->openParam.psSaveBufferSize / 1024));
+	}
+
+	VpuWriteReg(BIT_FRAME_MEM_CTRL,
 			    ((pDecInfo->openParam.chromaInterleave << 1) |
 			     IMAGE_ENDIAN));
-	} else {
-		if (pCodecInst->codecMode == AVC_DEC) {
-			VpuWriteReg(CMD_DEC_SEQ_PS_BB_START,
-					pDecInfo->openParam.psSaveBuffer);
-			VpuWriteReg(CMD_DEC_SEQ_PS_BB_SIZE,
-				(pDecInfo->openParam.psSaveBufferSize / 1024));
-		}
-	}
 
 	VpuWriteReg(CMD_DEC_SEQ_SRC_SIZE, pDecInfo->picSrcSize);
 
-- 
1.8.0

